packages/renderer/src/renderer.ts

```diff
@@ -72,40 +72,38 @@ export class Renderer implements IAgentPlugin {
     context?: IRendererContext
   ): Promise<IRenderResult> {
     try {
-      const renderMethods: RenderMethodPayload[] | [] = [];
-
-      // const [expandedDocument] = await expandVerifiableCredential(
-      //   args.credential
-      // );
-      // const renderMethods: RenderMethodPayload[] | [] =
-      //   extractRenderMethods(expandedDocument);
-
-      // TODO: There's an issue with W3 availability causing the fetching of some W3 context files to fail. Since we know the exact location of the template we can bypass the JSONLD expansion. This is a temporary workaround.
+     
+      const [expandedDocument] = await expandVerifiableCredential(
+        args.credential
+      );
+    
+      const renderMethods: RenderMethodPayload[] | [] =
+        extractRenderMethods(expandedDocument);
 
-      const render = args.credential.render;
-      if (!Array.isArray(render)) {
+      if (!Array.isArray(renderMethods)) {
         throw new Error('Render method not found in the verifiable credential');
       }
-
-      const template = render[0]?.template;
-      const type = render[0]?.['@type'];
-      if (!template || !type) {
-        throw new Error('Render method must have template and @type.');
-      }
-
-      renderMethods[0] = {
-        template,
-        '@type': type,
-      };
-
+      
       const documents = await Promise.all(
         renderMethods.map(async (renderMethod) => {
           const rendererProvider = this.getProvider(renderMethod['@type']);
           const document = await rendererProvider.renderCredential(
-            renderMethod.template,
-            args.credential
+            {
+              template: renderMethod.template,
+              document: args.credential,
+              url: renderMethod.url,
+              digestMultibase: renderMethod.digestMultibase,
+              mediaType: renderMethod.mediaType,
```

Comment:
The fields are specific to render template 2024 that should be in the render template 2024 provider